src/js/components/DataTable/Header.js

```diff
@@ -51,6 +58,24 @@ const Header = ({
           />
         )}
 
+        {(selected || onSelect) && (
+          <TableCell>
+            <CheckBox
+              checked={selected.length === data.length}
+              indeterminate={
+                selected.length > 0 && selected.length < data.length
+              }
+              onChange={() => {
+                if (selected.length > 0) onSelect([]);
```

Comment:
Not sure I understand the logic here, can you elaborate?

Reply:
I've added some comments in the code